### PR TITLE
Allow custom endpoints for all REST models

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Allow custom endpoints for all REST models. Fixes https://github.com/explosion/spacy-llm/issues/387.

### Corresponding documentation PR

https://github.com/explosion/spaCy/pull/13169

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran all tests in `tests` and `usage_examples/tests`, and all new and existing tests passed. This includes
  - all external tests (i. e. `pytest` ran with `--external`)
  - all tests requiring a GPU (i. e. `pytest` ran with `--gpu`)
- [ ] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy_llm/models/rest/anthropic/registry.py [hunk 1]
```diff
@@ -27,12 +28,13 @@ def anthropic_claude_2(
     interval (float): Time interval (in seconds) for API retries in seconds. We implement a base 2 exponential backoff
         at each retry.
     max_request_time (float): Max. time (in seconds) to wait for request to terminate before raising an exception.
+    endpoint (str): Endpoint to use.. Defaults to standard endpoint.
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:31]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/anthropic/registry.py [hunk 2]
```diff
@@ -61,12 +64,13 @@ def anthropic_claude_1(
...
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:67]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/anthropic/registry.py [hunk 3]
```diff
@@ -97,12 +102,13 @@ def anthropic_claude_instant_1(
...
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:105]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/anthropic/registry.py [hunk 4]
```diff
@@ -133,12 +140,13 @@ def anthropic_claude_instant_1_1(
...
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:143]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/anthropic/registry.py [hunk 5]
```diff
@@ -167,12 +176,13 @@ def anthropic_claude_1_0(
...
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:179]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/anthropic/registry.py [hunk 6]
```diff
@@ -201,12 +212,13 @@ def anthropic_claude_1_2(
...
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:215]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/anthropic/registry.py [hunk 7]
```diff
@@ -235,12 +248,13 @@ def anthropic_claude_1_3(
...
```
Comment (svlandeg) [inline spacy_llm/models/rest/anthropic/registry.py:251]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

File: spacy_llm/models/rest/cohere/registry.py [hunk 8]
```diff
@@ -29,11 +30,12 @@ def cohere_command(
     interval (float): Time interval (in seconds) for API retries in seconds. We implement a base 2 exponential backoff
         at each retry.
     max_request_time (float): Max. time (in seconds) to wait for request to terminate before raising an exception.
+    endpoint (str): Endpoint to use. Defaults to standard endpoint.
```
Comment (svlandeg) [inline spacy_llm/models/rest/cohere/registry.py:33]:
```suggestion
    endpoint (Optional[str]): Endpoint to use. Defaults to standard endpoint.
```

**Review by svlandeg — COMMENTED** [review]
Makes sense!

File: spacy_llm/models/rest/palm/registry.py [hunk 9]
```diff
@@ -27,13 +28,15 @@ def palm_bison(
...
     """
     return PaLM(
         name=name,
-        endpoint=Endpoints.TEXT.value
-        if name in {"text-bison-001"}
-        else Endpoints.MSG.value,
+        endpoint=endpoint
+        or (
+            Endpoints.TEXT.value if name in {"text-bison-001"} else Endpoints.MSG.value
+        ),
```
[the remark this answers is not in the document]
Comment (rmitsch) [reply spacy_llm/models/rest/palm/registry.py]: Fixed in https://github.com/explosion/spacy-llm/pull/390/commits/48a8571adf937e9628a6d1fa832bd9fa53f81640.

**Pull request merged**